doc/guides/tfm/testsuites.rst

```diff
@@ -0,0 +1,63 @@
+Test Suites
+###########
+
+TF-M includes two sets of test suites:
+
+* tf-m-tests - Standard TF-M specific regression tests
+* psa-arch-tests - Test suites for specific PSA APIs (secure storage, etc.)
+
+These test suites can be run from Zephyr via an appropriate sample application
+in the samples/tfm_integration folder.
+
+TF-M Regression Tests
+*********************
+
+The regression test suite can be run via the ``tfm_regression_test`` sample,
+for example:
+
+   .. code-block:: bash
+
+     $ west build samples/tfm_integration/tfm_regression_test/ \
+       -p -b mps2_an521_ns -t run
+
+This sample tests various services and communication mechanisms across the
+NS/S boundary via the PSA APIs. They provide a useful sanity check for proper
+integration between the NS RTOS (Zephyr in this case) and the secure
+application (TF-M).
+
+PSA Arch Tests
+**************
+
+The PSA Arch Test suite, available via ``tfm_psa_test``, contains a number of
+test suites that can be used to validate that PSA API specifications are
+being followed by the secure application, TF-M being an implementation of
+the Platform Securtity Architecture (PSA).
+
+Only one of these suites can be run at a time, with the test suite set via one
+of the following kconfig options:
+
+* ``CONFIG_TFM_PSA_TEST_CRYPTO``
```

Comment:
sample.yaml only enables 2 of these. Our test-coverage could be improved here.

Are these run as part of CI, or do we have to move them to tests/ in order to do that?

Reply:
At least for Nordic devices Protected Storage and Internal Trusted Storage are the only suites supported, and they're run in our CI. Whether or not the other platforms support more suites I don't know, and I also don't know what's run in Zephyr CI.

Also FYI: The Storage suites is just a combination of the PS and ITS suites, so it doesn't add any value.

Reply:
I've left the list as is (all existing KConfig options are present), but do you have a specific proposal here? This seems like something that should be resolved in a distinct PR for the sample in question.

Reply:
If all of these options pass for the default test board then I think we should enable them in the sample.yaml file.
I was asking because I don't know the current state of which of these tests would actually pass at the moment.

Definitely something that can be addressed in a separate PR, I agree.